### DISCUSSION TITLE
Provide *pyi files

### DISCUSSION THREAD

**Discussion by ferrine** [pymc-devs/pytensor#86] [Ideas]
As in numpy https://github.com/numpy/numpy/blob/297c66131f2a32ebb7dfb0aeb9d88d917a791430/numpy/lib/arraysetops.pyi#L110
There is a plenty of stuff that could be improved for mypy. Multiple signatures and return types, clever type hints and introspection. We don't do that so far

**Comment by ferrine** [comment]
@michaelosthege

**Reply by michaelosthege** [reply]
I think this should be covered by our `py.typed` file: e464e50a2f3a664ef1303c91070e7598de008313

At least that's how I understood https://peps.python.org/pep-0561/#packaging-type-information

The presence of the `py.typed` file, and our existing inline type annotations should be all we need.

**Reply by michaelosthege** [reply]
Maybe the `.pyi` stubs support more than what we currently have? But maintaining them separate from the inline annotations sounds very cumbersome..

For the `@overload` example you linked, we can also fix this on the `.py` side by writing two functions (one having an `@overload` decorator).
